packages/components/date-picker/src/date-picker-com/panel-month-range.vue

```diff
@@ -255,7 +255,17 @@ export default defineComponent({
     // pickerBase.hub.emit('SetPickerOption', ['isValidValue', isValidValue])
     ctx.emit('set-picker-option', ['formatToString', formatToString])
     const pickerBase = inject('EP_PICKER_BASE') as any
-    const { shortcuts, disabledDate, format, defaultValue } = pickerBase.props
+    const { disabledDate, format, defaultValue } = pickerBase.props
+
+    const shortcuts = ref([])
+    shortcuts.value = pickerBase.props
+    watch(
+      () => pickerBase.props.shortcuts,
+      (val) => {
+        shortcuts.value = val
+      },
+      { immediate: true }
+    )
```

Comment:
Same happens here
I see these code in 3 different places, I think we should consider encapsulate the code to avoid code duplication. Please use a `useShortcut` to encapsulate it.